src/rgw/rgw_sal_motr.cc

```diff
@@ -1838,23 +1841,27 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
     ext.iv_vec.v_count[0] = bs;
     attr.ov_vec.v_count[0] = 0;
 
+    left -= actual;
     // Read from Motr.
     op = nullptr;
     rc = m0_obj_op(this->mobj, M0_OC_READ, &ext, &buf, &attr, 0, 0, &op);
     ldpp_dout(dpp, 20) << "MotrObject::read_mobj(): init read op rc=" << rc << dendl;
-    if (rc != 0)
+    if (rc != 0) {
+      ldpp_dout(dpp, 0) << __func__ << ": read failed during m0_obj_op, rc=" << rc << dendl;
       goto out;
+    }
     m0_op_launch(&op, 1);
     rc = m0_op_wait(op, M0_BITS(M0_OS_FAILED, M0_OS_STABLE), M0_TIME_NEVER) ?:
          m0_rc(op);
     m0_op_fini(op);
     m0_op_free(op);
-    if (rc != 0)
+    if (rc != 0) {
+      ldpp_dout(dpp, 0) << __func__ << ": read failed, m0_op_wait rc=" << rc << dendl;
       goto out;
-
+    }
     // Call `cb` to process returned data.
     ldpp_dout(dpp, 20) << "MotrObject::read_mobj(): call cb to process data" << dendl;
-    cb->handle_data(bl, off, actual);
+    cb->handle_data(bl, 0, actual);
```

Comment:
For byte-range GET, the offset `0` is not the correct offset in the bufferlist.
You can leave the test and fix of byte-range GET in a new PR.

Reply:
Shouldn't it have the original value of `off` here which was passed to `MotrObject::read_mobj()`?

Reply:
That is the point. We are always writing data received from the motr into `bl` starting from 0th index in every iteration. We can not read from any other offset. That data will discarded at the end of every iteration. That will in-turn make sure that we are not accumulating too much data in `bl`.

Reply:
Sorry, I don't understand. This is a reading function, not writing. Why we cannot read from any other offset than zero?

Reply:
Sining explained:
> 0 is the offset local to the bl , not the offset in the whole object